test/fw/ptl/utils/pbs_dshutils.py

```diff
@@ -1968,6 +1968,25 @@ def mkdtemp(self, hostname=None, suffix='', prefix='PtlPbs', dir=None,
                        sudo=True)
         return fn
 
+    def create_dyn_res_script(self, body, prefix=None, suffix=None, perm=0755,
+                              dirname=None, host=None):
+        """
+        Create a dynamic resource script owned by root
+        """
+        home_dir = os.path.expanduser("~")
+        fp = self.create_temp_file(prefix="mom_resc", suffix=".scr",
+                                   body=body, asuser="root",
+                                   dirname=home_dir, hostname=host)
+        self.chmod(path=fp, mode=perm, sudo=True, hostname=host)
```

Comment:
create_temp_file should probably just take a 'mode' argument, it does a chmod internally with a hardcoded 644, so it would be easy to just have it take the 'mode' argument, what do you think?